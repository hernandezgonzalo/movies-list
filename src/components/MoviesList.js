import React from "react";
import ListItem from "./ListItem";
import styled from "styled-components";

const Table = styled.table`
  border-collapse: collapse;

  th {
    background: #ccc;
  }

  th,
  td {
    border: 1px solid #ccc;
    padding: 8px;
  }

  tr:nth-child(even) {
    background: #efefef;
  }
`;

const MoviesList = ({ movies }) => {
  console.log(movies);
  return (
    <Table>
      <tbody>
        {movies.map((movie, index) => (
          <ListItem key={index} movie={movie} />
        ))}
      </tbody>
    </Table>
  );
};

export default MoviesList;
