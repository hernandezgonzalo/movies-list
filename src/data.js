// using sessionStorage as data base

const initState = {
  movies: [
    {
      title: "La vida de Brian",
      genres: ["comedy"],
      watched: false,
      index: 3
    },
    {
      title: "El viaje de Chihiro",
      genres: ["fantasy"],
      watched: false,
      index: 2
    },
    {
      title: "Ã‰rase una vez en... Hollywood",
      genres: ["drama", "comedy"],
      watched: false,
      index: 1
    }
  ]
};

const setDummyData = () => {
  Object.keys(initState).forEach(key => {
    sessionStorage.setItem(key, JSON.stringify(initState[key]));
  });
};

export default setDummyData;
