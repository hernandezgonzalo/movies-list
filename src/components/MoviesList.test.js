import React from "react";
import { shallow } from "enzyme";
import MoviesList from "./MoviesList";
import { initState } from "../data";

describe("MoviesList", () => {
  it("should render correctly with no props", () => {
    const component = shallow(<MoviesList />);

    expect(component).toMatchSnapshot();
  });

  it("should render correctly with props", () => {
    const component = shallow(<MoviesList movies={initState.movies} />);

    expect(component).toMatchSnapshot();
  });
});
