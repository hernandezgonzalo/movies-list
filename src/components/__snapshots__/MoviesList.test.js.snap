// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList should render correctly with no props 1`] = `
<styled.table>
  <tbody />
</styled.table>
`;

exports[`MoviesList should render correctly with props 1`] = `
<styled.table>
  <tbody>
    <Connect(ListItem)
      key="0"
      movie={
        Object {
          "genres": Array [
            "comedy",
          ],
          "index": 3,
          "title": "La vida de Brian",
          "watched": false,
        }
      }
    />
    <Connect(ListItem)
      key="1"
      movie={
        Object {
          "genres": Array [
            "fantasy",
          ],
          "index": 2,
          "title": "El viaje de Chihiro",
          "watched": false,
        }
      }
    />
    <Connect(ListItem)
      key="2"
      movie={
        Object {
          "genres": Array [
            "drama",
            "comedy",
          ],
          "index": 1,
          "title": "Ã‰rase una vez en... Hollywood",
          "watched": false,
        }
      }
    />
  </tbody>
</styled.table>
`;
